database-backup-postgres.sh

```
@@ -0,0 +1,207 @@
+    [[ -z "$schema" ]] && return
+    
+    log "Backing up schema: $schema to $filename"
+    
+    if [[ "$use_docker" == "true" ]]; then
+        docker exec -e PGPASSWORD="$pass" "$service" "$pg_dump_cmd" -U "$user" -h "$host" -p "$port" -d "$database" -n "$schema" > "${output_dir}/${filename}.sql"
+    else
+        PGPASSWORD="$pass" "$pg_dump_cmd" -U "$user" -h "$host" -p "$port" -d "$database" -n "$schema" -f "${output_dir}/${filename}.sql"
+    fi
+    
+    # Check if backup was successful
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)